lib/seahorn/BvOpSem2.cc

```diff
@@ -553,6 +555,32 @@ class OpSemVisitor : public InstVisitor<OpSemVisitor>, OpSemVisitorBase {
     setValue(*CS.getInstruction(), res);
   }
 
+  void visitAssertIf(CallSite CS) {
+    Expr ante = lookup(*CS.getArgument(0));
+    Expr conseq = lookup(*CS.getArgument(1));
+    m_ctx.m_z3_solver->reset();
+    m_ctx.m_z3_solver->assertExpr(ante);
+    boost::tribool res = m_ctx.m_z3_solver->solveAssuming(m_ctx.side());
```

Comment:
this seems wrong. `side` is a side-condition, not a literal.
I would expect
```
m_ctx.m_z3_solver->assertExpr(m_ctx.side());
m_ctx.m_z3_solver->solve();
```

Reply:
Fixed